src/EnergyPlus/EarthTube.cc

```diff
@@ -152,12 +157,53 @@ void GetEarthTube(EnergyPlusData &state, bool &ErrorsFound) // If errors found i
     // Following used for reporting
     state.dataEarthTube->ZnRptET.allocate(state.dataGlobal->NumOfZones);
 
-    constexpr std::string_view cCurrentModuleObject = "ZoneEarthtube";
-    int TotEarthTube = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+    std::string_view cCurrentModuleObject = "ZoneEarthtube:Parameters";
+    int totEarthTubePars = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+
+    state.dataEarthTube->EarthTubePars.allocate(totEarthTubePars);
+
+    for (Loop = 1; Loop <= totEarthTubePars; ++Loop) {
+        auto &thisEarthTubePars = state.dataEarthTube->EarthTubePars(Loop);
+        state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                 cCurrentModuleObject,
+                                                                 Loop,
+                                                                 state.dataIPShortCut->cAlphaArgs,
+                                                                 NumAlpha,
+                                                                 state.dataIPShortCut->rNumericArgs,
+                                                                 NumNumber,
+                                                                 IOStat,
+                                                                 state.dataIPShortCut->lNumericFieldBlanks,
+                                                                 state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                 state.dataIPShortCut->cAlphaFieldNames,
+                                                                 state.dataIPShortCut->cNumericFieldNames);
+
+        thisEarthTubePars.nameParameters = state.dataIPShortCut->cAlphaArgs(1);
+        // Check to make sure name is unique
+        for (int otherParams = 1; otherParams < Loop; ++otherParams) {
+            if (UtilityRoutines::SameString(thisEarthTubePars.nameParameters, state.dataEarthTube->EarthTubePars(otherParams).nameParameters)) {
+                ShowSevereError(state,
+                                format("{}: {} = {} is not a unique name.",
+                                       cCurrentModuleObject,
+                                       state.dataIPShortCut->cAlphaFieldNames(1),
+                                       state.dataIPShortCut->cAlphaArgs(1)));
+                ShowContinueError(state, format("Check the other {} names for a duplicate.", cCurrentModuleObject));
+                ErrorsFound = true;
+            }
+        }
```

Comment:
I don't think we can actually have unique names.  The input processor will fail early if the objects cannot be made into an internal JSON structure.  So this is not needed.  Perhaps I'm wrong...@mjwitte?